TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -19,6 +19,26 @@ namespace TypeCobol.Compiler.CupParser.NodeBuilder
     public class ProgramClassBuilder : IProgramClassBuilder
     {
         private static bool _SourceCodeDumped;
+        private static bool _AnomalousLineIndexTraced;
+
+        internal void LogAnomalousLineIndex()
+        {
+            const string message = "Anomalous line index found !";
+            if (_SourceCodeDumped)
+            {
+                // Full source with changes has already been dumped, just add a trace (once) to help notice the line index problem
+                if (!_AnomalousLineIndexTraced)
+                {
+                    LoggingSystem.LogMessage(LogLevel.Error, message);
```

Comment:
We need a full dump when a problem is found with index.
We cannot be sure that the dump made for invalid parent is done in the same instance of ProgramClassBuilder than line index problem.

Reply:
Now separating dumps on each cause